src/testers.rs

```diff
@@ -1,30 +1,123 @@
 use errors::{TeraResult, TeraError};
-use serde_json::value::{Map, Value};
-use parser::Node;
-
-use std::collections::LinkedList;
-
-// TODO: Don't expose the AST to tester functions.
-pub type TesterFn = fn(context: &Map<String, Value>,
-                       value: &Node,
-                       params: LinkedList<Value>)
-                       -> TeraResult<bool>;
-
-/// Returns true if `value` is defined in the given context. Otherwise, returns
-/// false.
-pub fn defined(context: &Map<String, Value>, value: &Node, params: LinkedList<Value>)
-        -> TeraResult<bool> {
-    if params.len() != 0 {
-        return Err(TeraError::TestError("defined".to_string(),
-            "defined should not be called with parameters".to_string()))
+use serde_json::value::{Value};
+use context::ValueNumber;
+
+
+
+pub type TesterFn = fn(
```

Comment:
This isn't idiomatic Rust style, and in this case, the idiomatic Rust style takes up just as many lines, so it's better to use that.